--- charch	2020-09-15 19:47:55.412365500 -0700
+++ charch_a	2020-09-15 19:43:41.985888800 -0700
@@ -106,12 +106,31 @@
 	fi
 }
 
+# Find the outermost mountpoint that charch is currently executing from
+find_mountpoint() {
+	local path=`readlink -f "$1"`
+	until mountpoint -q "$path" &>/dev/null
+	do
+		path=${path%/*}
+		path=${path:-/}
+
+		# Break if we reach the end of the filesystem
+		[[ "$path" == "/" ]] && ! mountpoint -q "/" &> /dev/null && break
+	done
+	echo "$path"
+}
+
 # Mount essentials and remound current mount as suid,exec
 setup_mounts() {
 	safe_mount_dir /dev tmpfs
 	safe_mount_dir /sys sysfs
 	safe_mount_dir /proc proc
 	safe_mount_file /etc/resolv.conf
+
+	# Remount data partition with suid to allow sudo to function
+	local root_mount=`find_mountpoint "$CHROOT_DIR"`
+	mount | awk '{print $3}' | grep -q "^$root_mount\$" &&
+		mount -o remount,suid,exec "$root_mount" &> /dev/null
 }
 
 # Enter the chroot with the provided command
